packages/geo/src/lib/workspace/shared/feature-workspace.service.ts

```diff
@@ -0,0 +1,117 @@
+import { Injectable } from '@angular/core';
+
+import {
+  ActionStore,
+  EntityTableTemplate,
+  EntityStoreFilterCustomFuncStrategy,
+  EntityRecord,
+  EntityStoreStrategyFuncOptions
+} from '@igo2/common';
+
+import {
+  FeatureStore,
+  FeatureStoreLoadingLayerStrategy,
+  FeatureStoreSelectionStrategy,
+  FeatureStoreInMapExtentStrategy,
+  Feature
+} from '../../feature';
+import { VectorLayer } from '../../layer';
+import { IgoMap } from '../../map';
+import { SourceFieldsOptionsParams } from '../../datasource';
+
+import { FeatureWorkspace } from './feature-workspace';
+import { FeatureActionsService } from './feature-actions.service';
+import { skipWhile, take } from 'rxjs/operators';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class FeatureWorkspaceService {
+
+  constructor(private featureActionsService: FeatureActionsService) {}
+
+  createWorkspace(layer: VectorLayer, map: IgoMap): FeatureWorkspace {
+    const wks = new FeatureWorkspace({
+      id: layer.id,
+      title: layer.title,
+      layer,
+      map,
+      entityStore: this.createFeatureStore(layer, map),
+      actionStore: new ActionStore([]),
+      meta: {
+        tableTemplate: undefined
+      }
+    });
+    this.createTableTemplate(wks, layer);
+    this.featureActionsService.loadActions(wks);
+    return wks;
+
+  }
+
+  private createFeatureStore(layer: VectorLayer, map: IgoMap): FeatureStore {
+    const store = new FeatureStore([], {map});
+    store.bindLayer(layer);
+
+    const loadingStrategy = new FeatureStoreLoadingLayerStrategy({});
+    const inMapExtentStrategy = new FeatureStoreInMapExtentStrategy({});
+    const selectionStrategy = new FeatureStoreSelectionStrategy({
+      map,
+      hitTolerance: 15
+    });
+    store.addStrategy(loadingStrategy, true);
+    store.addStrategy(inMapExtentStrategy, true);
+    store.addStrategy(selectionStrategy, true);
+    store.addStrategy(this.createFilterInMapExtentStrategy(), false);
+    return store;
+  }
+
+  private createTableTemplate(workspace: FeatureWorkspace,  layer: VectorLayer): EntityTableTemplate {
```

Comment:
I'm pretty sure the async nature of this function could be a problem if an entity-table is created before this method completes. You can leave it like this for now and get back to it if an issue arises.

Reply:
I will leave it like this and fix it hereafter.